proposals/nnnn-member-macro-conformances.md

```diff
@@ -0,0 +1,116 @@
+
+However, there are some times when the member used for the conformance really needs to be part of the original type definition. For example:
+
+- An initializer in a non-final class needs to be a `required init` to satisfy a protocol requirement.
+- An overridable member of a non-final class.
+- A stored property or case can only be in the primary type definition.
+
+For these cases, a member macro can produce the declarations. However, member macros aren't provided with any information about which protocol conformances they should provide members for, so a macro might erroneously try to add conforming members to a type that already conforms to the protocol (e.g., through a superclass). This can make certain macros---such as macros that implement the `Encodable` or `Decodable` protocols---unimplemented.
+
+## Proposed solution
+
+To make it possible for a member macro to provide the right members for a conformance, we propose to extend member macros with the same ability that extension macros have to reason about conformances. Specifically:
+
+* The `attached` attribute specifying a `member` role gains the ability to specify the set of protocol conformances it is interest in, the same way an `extension` macro specifies the conformances it can provide.
+* The `expansion` operation for a `MemberMacro` -conforming implementation receives the set of protocols that were stated (as above) and which the type does not already conform to.
+
+This information allows a macro to reason about which members it should produce to satisfy conformances. Member macros that are interested in conformances are often going to also be extension macros, which work along with the member macro to provide complete conformance information.
+
+As an example, consider a `Codable` macro that provides the `init(from:)` and `encode(to:)` operations required by the `Decodable` and `Encodable` protocols, respectively. Such a macro could be defined as follows:
+
+```swift
+@attached(member, conformances: Decodable, Encodable, names: named(init(from:), encode(to:)))
+@attached(extension, conformances: Decodable, Encodable, names: named(init(from:), encode(to:)))
+macro Codable() = #externalMacro(module: "MyMacros", type: "CodableMacro")
+```
+
+This macro has several important decisions to make about where and how to generate `init(from:)` and `encode(to:)`:
+
+* For a struct, enum, actor, or final class, `init(from:)` and `encode(to:)`should be emitted into an extension (via the member role) along with the conformance. This is both good style and, for structs, ensures that the initializer doesn't inhibit the memberwise initializer.
+* For a non-final class, `init(from:)` and `encode(to:)`should be emitted into the main class definition (via the member role) so that can be overridden by subclasses.
```

Comment:
```suggestion
* For a non-final class, `init(from:)` and `encode(to:)` should be emitted into the main class definition (via the member role) so that they can be overridden by subclasses.
```